lms/services/digest.py

```diff
@@ -1,11 +1,303 @@
+from dataclasses import dataclass
+from typing import Iterable, Optional, Tuple
+
+from sqlalchemy import func, select
+
+from lms.models import (
+    Assignment,
+    AssignmentGrouping,
+    AssignmentMembership,
+    Course,
+    Grouping,
+    LTIRole,
+    User,
+)
+from lms.services.h_api import HAPI
+from lms.services.mailchimp import EmailRecipient, EmailSender, MailchimpService
+
+
 class DigestService:
     """A service for generating "digests" (activity reports)."""
 
+    def __init__(self, db, h_api, mailchimp_service, sender):
+        self._db = db
+        self._h_api = h_api
+        self._mailchimp_service = mailchimp_service
+        self._sender = sender
+
     def send_instructor_email_digests(
         self, audience, updated_after, updated_before, override_to_email=None
     ):
         """Send instructor email digests for the given users and timeframe."""
+        annotations = self._h_api.get_annotations(
```

Comment:
Note that the signature of `send_instructor_email_digests()` (which is the service's only public method) hasn't changed. So the calling code (in the Celery task) doesn't need to change.